01-syntax/03-data-structures/deep-comparation/deepEqual.js

```diff
@@ -4,15 +4,49 @@
  */
 /* global deepEqual */
 
-// → Your code here.
-
 var obj = {
   here: {
     is: 'an'
   },
   object: 2
 };
 
+function deepEqual(firstObj, secondObj) {
+  var firstKey;
+  var secondKey;
+  var same;
+
+  arguments[0] = firstObj || {};
+  arguments[1] = secondObj || {};
+
+  if (Object.keys(firstObj).length !== Object.keys(secondObj).length) {
+    return false;
+  } else if (firstObj === secondObj) {
+    return true;
+  }
+
+  for (firstKey in firstObj) {
+    if (firstObj.hasOwnProperty(firstKey)) {
+      same = false;
+      for (secondKey in secondObj) {
+        if (secondObj.hasOwnProperty(secondKey) &&
+          typeof firstObj[firstKey] === 'object' &&
+          typeof secondObj[secondKey] === 'object' &&
+          firstKey === secondKey &&
+          firstObj[firstKey] && secondObj[secondKey]) {
+          same = deepEqual(firstObj[firstKey], secondObj[secondKey]);
+        } else if (firstKey === secondKey) {
+          same = (firstObj[firstKey] === secondObj[secondKey]);
+        }
+      }
+      if (!same) {
+        return false;
+      }
+    }
+  }
+  return true;
+}
+
 console.log(deepEqual(obj, obj));
```

Comment:
I have an additional personal BONUS task  for you

What about objects which contain arrays ? 
Could you extend your function to support nesting with arrays ?